TS_HW1/home_work_1.ts

```diff
@@ -1,112 +1,121 @@
 class School {
-  directions = [];
+  public directions: any[] = []; // має бути : Direction[]
 
-  addDirection(direction) {
+  public addDirection(direction: any): void {
     this.directions.push(direction);
   }
 }
 
 class Direction {
-  levels = [];
+  public levels: any[] = []; // : Level[]
+  private _name: string;
 
-  get name() {
+  get name(): string {
     return this._name;
   }
 
-  constructor(name) {
+  constructor(name: string) {
     this._name = name;
   }
 
-  addLevel(level) {
+  public addLevel(level: any): void {
     this.levels.push(level);
   }
 }
 
 class Level {
-  groups = [];
+  public groups: any[] = []; // : Group[]
+  private _name: string;
+  private _program: string;
 
-  constructor(name, program) {
-    this.name = name;
+  constructor(name: string, program: string) {
+    this._name = name;
     this._program = program;
   }
 
-  get name() {
+  get name(): string {
     return this._name;
   }
 
-  get program() {
+  get program(): string {
     return this._program;
   }
 
-  addGroup(group) {
+  public addGroup(group: any) {
     this.groups.push(group);
   }
 }
 
 class Group {
-  _students = [];
-
-  get students() {
+  private _students: any[] = []; // : Student[]
+  public directionName: string;
+  public levelName: string;
+  
+  get students(): any[] {
     return this._students;
   }
 
-  constructor(directionName, levelName) {
+  constructor(directionName: string, levelName: string) {
     this.directionName = directionName;
     this.levelName = levelName;
   }
 
-  addStudent(student) {
+  public addStudent(student: any) {
     this._students.push(student);
   }
 
-  showPerformance() {
-    const sortedStudents = this.students.toSorted(
-      (a, b) => b.getPerformanceRating() - a.getPerformanceRating()
+  public showPerformance() {
+    const sortedStudents: any[] = this.students.toSorted(
+      (a: any, b: any) => b.getPerformanceRating() - a.getPerformanceRating()
     );
 
     return sortedStudents;
   }
 }
 
 class Student {
-  grades = {};
-  attendance = [];
-
-  constructor(firstName, lastName, birthYear) {
+  public grades: any = {};            // певний об'єкт Grade ?
+  public attendance: boolean[] = [];  // або об'єкт ?
```

Comment:
This is definitely an array, but it doesn't matter which exactly type for now. "any[]" will be okay.